src/bin/flamegraph.rs

```diff
@@ -2,18 +2,31 @@ use std::fs::File;
 use std::io::{self, BufReader};
 use structopt::StructOpt;
 
-use inferno::flamegraph::{self, Options};
+use inferno::flamegraph::{self, Options, color::Palette};
 
 #[derive(Debug, StructOpt)]
 #[structopt(name = "inferno-flamegraph", author = "")]
 struct Opt {
     /// collapsed perf output file, or STDIN if not specified
     infile: Option<String>,
+    /// set color palette. choices are: hot, mem, io, wakeup, chain, java, js, perl, red, green, blue, aqua, yellow, purple, orange
+    #[structopt(long = "colors", default_value = "hot")]
+    colors: Palette,
+    /// colors are keyed by function name hash
+    #[structopt(long = "hash")]
+    hash: bool,
+    /// use consistent palette (palette.map)
```

Comment:
I _think_ `flamegraph.pl` also allows for specifying the file name to use?

Reply:
It does not: [pal_file](https://github.com/brendangregg/FlameGraph/blob/f857ebc94bfe2a9bfdc4f1536ebacfb7466f69ba/flamegraph.pl#L118) is initialized, and then directly used in [read_palette ](https://github.com/brendangregg/FlameGraph/blob/f857ebc94bfe2a9bfdc4f1536ebacfb7466f69ba/flamegraph.pl#L522)and [write_palette](https://github.com/brendangregg/FlameGraph/blob/f857ebc94bfe2a9bfdc4f1536ebacfb7466f69ba/flamegraph.pl#L513).
But we can add the feature as an optional argument for the `--cp` option ? Or a new option ?

Reply:
Ah, yes, looks like you're right! I think having `cp` optionally take an argument would be the best way to go, but sadly that [doesn't seem to be supported yet](https://github.com/clap-rs/clap/issues/1354), so let's stick with hard-coded `palette.map` for now!